From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Xu <peterx@redhat.com>
Date: Wed, 1 Apr 2026 16:28:38 -0400
Subject: [PATCH] vmstate: Rename VMS_NULLPTR_MARKER to VMS_MARKER_PTR_NULL
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Prepare for a new MARKER for non-NULL pointer.

Reviewed-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@futurfusion.io>
Reviewed-by: Fabiano Rosas <farosas@suse.de>
Reviewed-by: Philippe Mathieu-Daudé <philmd@linaro.org>
Signed-off-by: Peter Xu <peterx@redhat.com>
Reviewed-by: Juraj Marcin <jmarcin@redhat.com>
Link: https://lore.kernel.org/qemu-devel/20260401202844.673494-6-peterx@redhat.com
Signed-off-by: Fabiano Rosas <farosas@suse.de>
(cherry picked from commit 4ce421e8edfb25e87028a8050f0ba968f24ebed1)
---
 include/migration/vmstate.h | 2 +-
 migration/vmstate-types.c   | 6 +++---
 tests/unit/test-vmstate.c   | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/migration/vmstate.h b/include/migration/vmstate.h
index 73e1a52d0a..d0758f5b33 100644
--- a/include/migration/vmstate.h
+++ b/include/migration/vmstate.h
@@ -248,7 +248,7 @@ extern const VMStateInfo vmstate_info_uint64;
 extern const VMStateInfo vmstate_info_fd;
 
 /** Put this in the stream when migrating a null pointer.*/
-#define VMS_NULLPTR_MARKER (0x30U) /* '0' */
+#define VMS_MARKER_PTR_NULL          (0x30U)   /* '0' */
 extern const VMStateInfo vmstate_info_nullptr;
 
 extern const VMStateInfo vmstate_info_cpudouble;
diff --git a/migration/vmstate-types.c b/migration/vmstate-types.c
index 741a588b7e..32ef520821 100644
--- a/migration/vmstate-types.c
+++ b/migration/vmstate-types.c
@@ -342,10 +342,10 @@ static int get_nullptr(QEMUFile *f, void *pv, size_t size,
                        const VMStateField *field)
 
 {
-    if (qemu_get_byte(f) == VMS_NULLPTR_MARKER) {
+    if (qemu_get_byte(f) == VMS_MARKER_PTR_NULL) {
         return  0;
     }
-    error_report("vmstate: get_nullptr expected VMS_NULLPTR_MARKER");
+    error_report("vmstate: get_nullptr expected VMS_MARKER_PTR_NULL");
     return -EINVAL;
 }
 
@@ -354,7 +354,7 @@ static int put_nullptr(QEMUFile *f, void *pv, size_t size,
 
 {
     if (pv == NULL) {
-        qemu_put_byte(f, VMS_NULLPTR_MARKER);
+        qemu_put_byte(f, VMS_MARKER_PTR_NULL);
         return 0;
     }
     error_report("vmstate: put_nullptr must be called with pv == NULL");
diff --git a/tests/unit/test-vmstate.c b/tests/unit/test-vmstate.c
index 64d7b304ae..7e6e81738d 100644
--- a/tests/unit/test-vmstate.c
+++ b/tests/unit/test-vmstate.c
@@ -583,7 +583,7 @@ static void test_arr_ptr_str_no0_load(void)
 
 static uint8_t wire_arr_ptr_0[] = {
     0x00, 0x00, 0x00, 0x00,
-    VMS_NULLPTR_MARKER,
+    VMS_MARKER_PTR_NULL,
     0x00, 0x00, 0x00, 0x02,
     0x00, 0x00, 0x00, 0x03,
     QEMU_VM_EOF
